### ISSUE TITLE
Modbus updates

### ISSUE THREAD

**Issue by NubeDev** [NubeIO/flow-framework#218]
## register grouping

https://github.com/NubeIO/nubeio-rubix-app-modbus-py/blob/master/src/services/polling/modbus_polling.py#L135-L165
this is the packing and unpacking for the actual poll

https://github.com/NubeIO/nubeio-rubix-app-modbus-py/blob/26500edd6e87a41c50ece6a1c0ac0369484c03fb/src/services/polling/poll.py#L22 (edited)

**Comment by CanuckMarc** [comment]
### Modbus Point Polling Options
@NubeDev  as discussed we should consider adding additional options for tuning the modbus writes.  These might need better titles, but i'll describe their functionality below.  Only one of these settings can be selected for each point (so we should change to an enum selection)

**Read Only** - Don't ever write. 

**Write Only Once** - Currently it's partially implemented (https://github.com/NubeIO/rubix-point-server/issues/398).  The intent behind this setting is that when the write value is changed on the modbus master (platform), the value will be written to the modbus device; following this write (on `write` value COV) the point will continue to be polled with a `read` request.  The intent is that the value written to the modbus device could be changed by another modbus master, or by the program running on the modbus slave device.  Our Master should not re-write the value unless the write value is changed (or by a `force write` trigger). 

**Write to Maintain** - Points with this setting would write values to the modbus device if the `write` value changes, OR if the `read` value does not match the `write` value.  The intent here is to reduce the number of `write` commands, while still maintaining the correct setting value on the modbus device. 

**Write Always** - This is the current `write` strategy when `write only once` isn't selected.  It just writes the `write` value on every poll cycle.  This is too many writes for most cases. 


Additionally, we should consider implementing `fast`, `normal`, and `slow` polling options for tuning.  This would affect `read` polling times, and `write` cycle time for the `Write Always` option.

**Comment by Shiny380** [comment]
**Write Only Once** should **NOT** enter read mode after writing. this is the intent of **Write to Maintain**.
if there is a difference I didn't see then **Write Only Once** needs to be split into 2 different modes to allow this purely write only mode

**Comment by CanuckMarc** [comment]
`WRITE ONLY ONCE` and `WRITE TO MAINTAIN`, as described above, provide 2 seperate functionalities, both are required.  We may need to implement another mode that doesn't have any `READ` mode.  I'll add `WRITE ONLY` mode in my comment above.

**Comment by Shiny380** [comment]
A better modes description:
- **Write Always**
- **Write Once**
- **Write then Read** (updates with the read value)
- **Write with Override** (reads and overrides current values if doesn't match write)

If this wording without the parentheses isn't clear then they need to be re-worded so anyone can understand the modes without an explanation

**Comment by Shiny380** [comment]
moving this to a new issue #284
